### ISSUE TITLE
Self hosted Debian repository

### ISSUE THREAD

**Issue by avivace** [gbdev/rgbds#918]
Since we still didn't get enough attention from DDs, we can set up our own Debian repository to serve the rgbds package. I have some experience (and I maintain my own personal repository with custom builds) with https://www.aptly.info/ . I'd like to try to set up an APT repository just for RGBDS.

- Are we actually building `deb` packages from this repository when a new tag/release is out? If yes, does this happen automatically (i.e. CI?)
- Are changelogs being consistently written?

**Comment by ISSOtm** [comment]
No, nothing is currently being done. "All" that there is, is [the `rgbds-deb` repo](https://github.com/ISSOtm/rgbds-deb), which has usage instructions [in its wiki](https://github.com/ISSOtm/rgbds-deb/wiki/Process); and really, that's everything that's needed.

We can wire this to our CI, including putting the `debian/` stuff in a separate branch, and generating a `.deb` for each release once the changelog has been written. If we want to auto-generate it instead, we need to decide what to put i it.

**Comment by avivace** [comment]
> No, nothing is currently being done. "All" that there is, is [the `rgbds-deb` repo](https://github.com/ISSOtm/rgbds-deb), which has usage instructions [in its wiki](https://github.com/ISSOtm/rgbds-deb/wiki/Process); and really, that's everything that's needed.
> 
> We can wire this to our CI, including putting the `debian/` stuff in a separate branch, and generating a `.deb` for each release once the changelog has been written. If we want to auto-generate it instead, we need to decide what to put in it.

Great, thank you for the pointer. Will take a look as soon as I have a couple of minutes to try the deb building pipeline.

**Comment by avivace** [comment]
@ISSOtm Did we already have a discussion about adding the debian folder in the main repository? I see a lot of mainstream project doing that

**Comment by ISSOtm** [comment]
No, I don't think so. I'd be fine with it, tbh.

**Comment by robbi-blechdose** [comment]
> what's the eventual plan for these multiple repos? Can gbdev/rgbds-deb be deleted?
In my opinion yes. The Debian games team, under whose umbrella the repo is, requested that it's maintained on salsa. You could maintain the mirror here and on salsa, but I see no benefit to doing so.

> Is the progress in games-team/rgbds meant to be migrated back into the main gbdev/rgbds repo, or will we long-term just have to port the gbdev/rgbds commits over to this debian.org fork? Should we update the [RELEASE.md](https://github.com/gbdev/rgbds/blob/master/RELEASE.md) process once all this is done to document some steps needed for getting new releases into Debian as well?
I asked about this, the general consensus is that the packaging repo should be kept separate from the main repo.

**Comment by Rangi42** [comment]
RGBDS is now available in Debian unstable (https://packages.debian.org/sid/rgbds), and should be stable in Debian 14.

**Comment by ISSOtm** [comment]
Since RGBDS will be updated *very infrequently* there, we should nonetheless consider setting up a PPA that people can pull from. Especially since #1897 was merged, we now have access to https://cmake.org/cmake/help/v4.3/cpack_gen/deb.html which should enable us to offer `.deb` packages with each (new) release. We could also upload those to... well, wherever. That's the part I have less experience with. (Amusingly, [Codeberg has built-in support](https://forgejo.org/docs/latest/user/packages/debian/) for Debian repos! Migration when? 😜[^codeberg])

[^codeberg]: Actually, it looks like we could merely publish to Codeberg, since [we have an account there](https://codeberg.org/gbdev), and publishing packages there is a mere `curl` with a PAT. Hmm...
